src/api_endpoints.py

```diff
@@ -84,25 +170,65 @@ def post_results(self, result_json_obj):
         Post the result object produced while Result Generation
         using POST /results
         '''
+        headers = self.prepare_header()
         endpoint = self.url + self.result_endpoint
 
-        response = requests.post(endpoint, json=result_json_obj)
+        response = requests.post(
+            endpoint,
+            json=result_json_obj,
+            headers=headers)
 
         print("Status of post result response: ", response.status_code, '\n')
 
         return response.status_code
 
+    def post_workflow_and_save_response(self, workflow_path, response_path):
+        '''
+        Post the workflow and saves the response
+        '''
+        # Read the workflow json
+        with open(workflow_path, 'r') as f:
```

Comment:
with open(workflow_path) as f:
    json_data = json.load(f)

why not load directly using json

Reply:
Made suggested changes